aws-serverless-java-container-core/src/main/java/com/amazonaws/serverless/proxy/internal/serialization/AwsProxyRequestConverter.java

```diff
@@ -0,0 +1,54 @@
+package com.amazonaws.serverless.proxy.internal.serialization;
+
+import java.util.Map;
+import com.amazonaws.serverless.proxy.model.AwsProxyRequest;
+import com.amazonaws.serverless.proxy.model.Headers;
+import com.amazonaws.serverless.proxy.model.MultiValuedTreeMap;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.fasterxml.jackson.databind.util.Converter;
+
+/**
+ * Populates multi-valued collections from single-valued collections if the former are not given
+ * or are empty and the latter are present and populated. This is a real-life scenario when
+ * an ALB targets a Lambda function and `lambda.multi_value_headers.enabled` is not set to `true`
+ * on the target group. 
+ */
+public class AwsProxyRequestConverter implements Converter < AwsProxyRequest, AwsProxyRequest > {
+  @Override
+  public AwsProxyRequest convert(AwsProxyRequest value) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 23-29, lines 31-40 into separate methods.

Reply:
Good advice, and taken!